components/camel-firebase/src/test/java/org/apache/camel/component/firebase/FirebaseConsumerDeleteTest.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.firebase;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static junit.framework.TestCase.fail;
+
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.firebase.data.FirebaseMessage;
+import org.apache.camel.component.firebase.data.Operation;
+import org.apache.camel.component.firebase.provider.ConfigurationProvider;
+import org.apache.camel.component.firebase.provider.SampleInputProvider;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+/**
+ * Starts a route which listens to the remove event in Firebase. It then writes and deletes an entry in Firebase and
+ * asserts, if the entry was deleted or not.
+ */
+public class FirebaseConsumerDeleteTest {
```

Comment:
For the all the tests, since we need credentials, I suggest you to add a profile to the component POM to run the tests, if the profile is not activated then the tests will be skipped.